front/app/components/Form/Components/Controls/MapInput/Desktop/DesktopTabletView.tsx

```diff
@@ -0,0 +1,237 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+
+import Graphic from '@arcgis/core/Graphic';
+import Layer from '@arcgis/core/layers/Layer';
+import MapView from '@arcgis/core/views/MapView';
+import { Box, colors } from '@citizenlab/cl2-component-library';
+import { ControlProps } from '@jsonforms/core';
+import { Point } from 'geojson';
+import { useTheme } from 'styled-components';
+
+import { IMapConfig } from 'api/map_config/types';
+
+import useLocale from 'hooks/useLocale';
+
+import EsriMap from 'components/EsriMap';
+import ResetMapViewButton from 'components/EsriMap/components/ResetMapViewButton';
+import { Option } from 'components/UI/LocationInput';
+
+import { useIntl } from 'utils/cl-intl';
+import { sanitizeForClassname } from 'utils/JSONFormUtils';
+
+import ErrorDisplay from '../../../ErrorDisplay';
+import InstructionAnimation from '../components/InstructionAnimation';
+import LocationTextInput from '../components/LocationTextInput';
+import RemoveAnswerButton from '../components/RemoveAnswerButton';
+import UndoButton from '../components/UndoButton';
+import {
+  clearPointData,
+  handleMapClickMultipoint,
+  handleMapClickPoint,
+  checkCoordinateErrors,
+  updateDataAndDisplay,
+  handlePointDrag,
+  getInitialMapCenter,
+  MapInputType,
+  isLineOrPolygonInput,
+} from '../utils';
+
+type Props = {
+  mapConfig?: IMapConfig;
+  inputType: MapInputType;
+  mapLayers?: Layer[];
+  onMapInit?: (mapView: MapView) => void;
+  mapView?: MapView | null;
+  handleSinglePointChange: (point: GeoJSON.Point | undefined) => void;
+  handleMultiPointChange?: (points: number[][] | undefined) => void;
+
+  didBlur: boolean;
+};
+
+const DesktopView = ({
+  data,
+  path,
+  inputType,
+  errors,
+  mapConfig,
+  mapLayers,
+  onMapInit,
+  handleSinglePointChange,
+  handleMultiPointChange,
+  didBlur,
+  mapView,
+  id,
+  ...props
+}: ControlProps & Props) => {
+  const theme = useTheme();
+  const locale = useLocale();
+  const { formatMessage } = useIntl();
+  const [address, setAddress] = useState<Option>({
+    value: '',
+    label: '',
+  });
+  const layerCount = mapConfig?.data?.attributes?.layers?.length || 0;
+
+  // Create refs for custom UI elements
+  const resetButtonRef: React.RefObject<HTMLDivElement> = React.createRef();
+  const undoButtonRef: React.RefObject<HTMLDivElement> = React.createRef();
+  const instructionRef: React.RefObject<HTMLDivElement> = React.createRef();
+
+  // Create refs for dragging/editing a user's points on the map
+  const pointBeingDragged = useRef<Graphic | null>(null);
```

Comment:
Do these have to be refs? All these refs are making it really hard to understand what is happening, and how reactivity works. Did this not work with `useState`?